php-checks/src/main/java/org/sonar/php/checks/UnusedPrivateMethodCheck.java

```diff
@@ -46,10 +52,13 @@ public class UnusedPrivateMethodCheck extends PHPVisitorCheck {
 
   private final List<String> stringLiterals = new ArrayList<>();
   private final List<String> dynamicUsedMethods = new ArrayList<>();
+  private final List<String> methodsUsedInFirstClassCallables = new ArrayList<>();
 
   @Override
   public void visitClassDeclaration(ClassDeclarationTree tree) {
     stringLiterals.clear();
+    dynamicUsedMethods.clear();
```

Comment:
Clearing of this collection was also missing; it could lead to recognizing names mentioned in other classes' docs.

Reply:
Good catch!
It could be worth creating a test case ensuring this, which analyzes two files consecutively and would fail on any of these `.clear()` missing.